src/main/java/org/folio/mosaic/service/OrdersService.java

```diff
@@ -15,16 +13,24 @@ public class OrdersService {
 
   private final OrdersClient ordersClient;
   private final ConfigurationService configurationService;
-
-  public String createOrder(UUID templateId, CompositePurchaseOrder compositePurchaseOrder) {
-    log.info("createOrder:: Creating mosaic order with number: {}", compositePurchaseOrder.getPoNumber());
-
-    var orderTemplateId = templateId != null
-      ? templateId.toString()
+  private final MosaicOrderConverter orderConverter;
+
+  /**
+   * Creates an order using the new MosaicOrderRequest format.
+   * This method handles mapping the request to a CompositePurchaseOrder, applying
+   * template values, and creating the order in FOLIO.
+   */
+  public String createOrder(MosaicOrderRequest orderRequest) {
+    var requestOrderTemplateId = orderRequest.getOrderTemplateId();
+    var mosaicOrder = orderRequest.getMosaicOrder();
+    log.info("createOrder:: Creating mosaic order with title: {}", orderRequest.getMosaicOrder().getTitle());
+
+    String orderTemplateId = requestOrderTemplateId != null
+      ? requestOrderTemplateId
```

Comment:
orderTemplateId also would be useful in logs

Reply:
added